### PR TITLE
Added a test for custom query on a relation in an interface

### PR DESCRIPTION
**Pull request by EdwinDayot**
See https://github.com/rebing/graphql-laravel/issues/421

### CODE REVIEW

**Comment by mfn** [comment]
Can't consider merging until everything is green

**Comment by EdwinDayot** [comment]
Sure, it will require to take in consideration each `Type` implementing an `Interface` to take the right `model` and `relation`. I was thinking about some kind of `Closure` taking `model` as parameter once it is required, but I really don't know how to implement this. As I've no idea how to solve this at the moment, would you recommend a temporary solution to make this mergeable ?

I could write tests returning green, but I'd prefer it not to be forgotten

File: tests/Database/SelectFields/InterfaceTests/InterfaceTest.php [hunk 1]
```diff
@@ -47,6 +48,58 @@ public function testGeneratedSqlQuery(): void
         $this->assertSame($expectedResult, $result);
     }
 
+    public function testGeneratedRelationSqlQuery()
```
Comment (mfn) [inline tests/Database/SelectFields/InterfaceTests/InterfaceTest.php]: Please always add the return type (`: void` in this case), like we do it everywhere, thank you!

File: tests/Database/SelectFields/InterfaceTests/InterfaceTest.php [hunk 2]
```diff
@@ -47,6 +48,58 @@ public function testGeneratedSqlQuery(): void
...
+                'title' => 'Title of the comment',
+                'post_id' => $post->id,
+            ]);
+
+        $graphql = <<<'GRAPHQL'
+{
+  exampleInterfaceQuery {
+    id
+    title
+    exampleRelation{
```
Comment (mfn) [inline tests/Database/SelectFields/InterfaceTests/InterfaceTest.php]: Space ` ` before `{`

**Comment by mfn** [comment]
> I could write tests returning green, but I'd prefer it not to be forgotten

Oh, I see.

Well, a red test is unfortunately not the right approach to "keep track" of it. First: can't merge PRs not green, a good precaution.

Second: future PRs may or may not break stuff and then it's harder to segregate these issues.

Having a test showing "something is broke" or "not working" is totally fine! You just have to use other means to remind about, like an open bug report referencing this PR, etc.

**Comment by mfn** [comment]
Thank you very much for the test 👍

**Pull request merged**